File: arr문제.c
#include <stdio.h>

int main() {
	int arr[]={2,3,0,0,4,0,6};
	int n = sizeof(arr)/sizeof(arr[0]);
	
	int j=0;
	for(int i=0;  i < n ; i++){
	    if(arr[i]!=0){
	        arr[j]=arr[i];
	        j+=1;
	    }
	}
	
	for(int i=j; i< n ; i++){
	    arr[i]=0;
	}
	
	for(int i=0; i < n ; i++){
	    printf("%d",arr[i]);
	}
	return 0;
}

File: arr문제.py
def recoder(arr,n):
    j=0
    for i in range(n):
        if arr[i]!=0:
            arr[j]=arr[i]
            j+=1
    for i in range(j,n):
        arr[i]=0
#code`
arr=[6,0,8,2,3,0,4,0,1]
n = len(arr)
recoder(arr,n)
print(arr)